### PR TITLE
Support for 'id_token_hint'

### LINKED ISSUE #19355: id_token_hint param is rejected in PAR request
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

authentication

### Describe the bug

id_token_hint is an optional request param on the auth endpoint, as defined in the oidc spec.
It is accepted and available in authenticators when using the standard keycloak /auth endpoint.
However, when using PAR, this parameter is rejected because it is not listed as a known parameter (which it should be as it is defined in the oidc core specification) and therefore limited to 200 chars, which is too little considering an id_token_hint is a signed jwt.

### Version

18.0.2

### Expected behavior

Regardless whether using the standard auth endpoint or the PAR endpoint, request params as defined in the specs, should be available in the session during authentication flow processing.

### Actual behavior

id_token_hint is not available as it is rejected because not seen as a known parameter and therefore limited to 200 chars.

### How to Reproduce?

ad a real encoded id_token as id_token_hint (should be more then 200 chars) to a PAR request.

### Anything else?

Fix to apply:

[https://github.com/keycloak/keycloak/blob/aa9744246c634a408ac3fa30fb8d0f2d82aee060/services/src/main/java/org/keycloak/pr
otocol/oidc/endpoints/request/AuthzEndpointRequestParser.java#L78](https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/protocol/oidc/endpoints/request/AuthzEndpointRequestParser.java#L78)

Add a line to set the id_token_hint if it was added to the request

allready exists: OIDCLoginProtocol.ID_TOKEN_HINT
property idTokenHint to add in request object: https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/protocol/oidc/endpoints/request/AuthorizationEndpointRequest.java

### PR DESCRIPTION
**Pull request by arso**
This change includes a support for 'id_token_id' optional param to avoid situation when it is being trimmed to max length defined for unknown params. 

**Work in progress**:  integration test and better verification still needed.

Closes #19355

### CODE REVIEW

**Review by mposolda — COMMENTED** [review]
@arso Thanks for the PR! I've added only one inline note. Can you please check it?

File: services/src/main/java/org/keycloak/protocol/oidc/endpoints/AuthorizationEndpoint.java [hunk 1]
```diff
@@ -321,6 +321,9 @@ private void updateAuthenticationSession() {
         if (request.getMaxAge() != null) authenticationSession.setClientNote(OIDCLoginProtocol.MAX_AGE_PARAM, String.valueOf(request.getMaxAge()));
         if (request.getUiLocales() != null) authenticationSession.setClientNote(LocaleSelectorProvider.CLIENT_REQUEST_LOCALE, request.getUiLocales());
 
+        // TODO: what about clientNote? should it be set as well? (with my limited understanding of the domain I say 'no', but loginHint param triggers setting clientNote only)
+        if (request.getIdTokenHint() != null) authenticationSession.setAuthNote(OIDCLoginProtocol.ID_TOKEN_HINT, request.getIdTokenHint());
```
[the remark this answers is not in the document]
Comment (arso) [reply services/src/main/java/org/keycloak/protocol/oidc/endpoints/AuthorizationEndpoint.java:325]: You're right about `clientNote` - now it makes sense. Thank you!
Not setting `authNote` do not have any side effects from what  I can tell.

**Comment by arso** [comment]
I need little help with idea on how to prepare test case covering this scenario. I failed to make any integration test to fail with error when using `id_token_hint` (as described in the reported issue).  I was only able to verify that PAR requests params do get parsed correctly with this PR (or trimmed when `id_token_hint` was too long before the PR)  ... but that was via a debugger.

**Comment by mposolda** [comment]
> I need little help with idea on how to prepare test case covering this scenario. I failed to make any integration test to fail with error when using `id_token_hint` (as described in the reported issue). I was only able to verify that PAR requests params do get parsed correctly with this PR (or trimmed when `id_token_hint` was too long before the PR) ... but that was via a debugger.

@arso Ideally, for the issues like this, it is good to do something like:
- Write the test, which is testing the mentioned scenario and is failing
- Then fix the code
- Make sure that the test from above is passing (after fix is done)

If the test is not failing without your fix, it means that it is either:
- Incorrect test (it doesn't correctly test scenario from the bug)
- Issue doesn't exists anymore (Maybe it was already fixed)

In other words, it is important to reproduce the bug, which you are fixing (before start to fixing it). Ideally with automated test. If your automated test is passing even without your changes in the code, it means that it is either incorrect or issue doesn't exists anymore. So it may need to be investigated which of those is the case.

File: services/src/main/java/org/keycloak/protocol/oidc/endpoints/AuthorizationEndpoint.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mposolda) [reply services/src/main/java/org/keycloak/protocol/oidc/endpoints/AuthorizationEndpoint.java]: Resolving this conversation. Thanks!

**Comment by mposolda** [comment]
I've tried the test in my environment and I can indeed see that test is passing even without any changes in the code. This can  imply that it is possibly not correct test? I am closing this PR for now, but feel free to re-open if you want to continue investigating this and writing correct test for this issue.
